Cookbook for Sonoff Light Switches

bensuffolk opened the pull request:
Cookbook for various Sonoff Light Switch options including Basic, Dual and T1-3

## Description:

Additional cookbook documentation for Sonoff Lights switches

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x ] Link added in `/index.rst` when creating new documents for new components or cookbook.

bensuffolk committed:
Cookbook for Sonoff Light Switches

Cookbook for various Sonoff Light Switch options including Basic, Dual and T1-3

OttoWinter commented on cookbook/sonoff-basic-light-switch.rst:
```diff
@@ -0,0 +1,161 @@
+Background
+----------
+
+I was looking for a simple, cost effective, way to upgrade my lights so they could be controlled by
```
The ESPHome docs are not using the personal form "I" - with the exception of the changelog, but that has an author set in the SEO tags. Besides, the reader can't know who "I" refers to in this case. If it's somewhere else too that's a mistake.

Please change this to general (I don't know what to call it) style. Example:

-> The idea of this project is to find a simple, cost effective, way to [...]

bensuffolk replied:
Of course, my mistake. I will correct that tomorrow.

bensuffolk replied:
I made those changes a while ago now for you, any other changes you want to make this PR happen?

bensuffolk committed:
Removed references to myself

All references to I have been removed and re worded into a more generic text

bensuffolk committed:
Fixed Formatting issues to Pass Travis Checks

OttoWinter commented on cookbook/sonoff-basic-light-switch.rst:
```diff
@@ -0,0 +1,176 @@
+In the above code block, there is a *secrets.yaml* file so that you have just one place to change wifi
+details for all your devices.
+
+The logger level INFO is required to overcome a bug that can stop the Sonoff Basic connecting to the wifi.
```
```suggestion
```

This has been fixed since the creation of this PR :) https://github.com/esphome/issues/issues/455

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            condition:
+              switch.is_off: relay_2
+            then:
+              - switch.turn_on: relay_1_on
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x01, 0xA1]
```

These lines can be inlined now - there's a uart.write action now.

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            then:
+              - switch.turn_on: relay_1_on
+            else:
+              - switch.turn_on: relay_12_on
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x03, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            condition:
+              switch.is_off: relay_2
+            then:
+              - switch.turn_on: relay_12_off
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x00, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            then:
+              - switch.turn_on: relay_12_off
+            else:
+              - switch.turn_on: relay_2_on
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x02, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            condition:
+              switch.is_off: relay_1
+            then:
+              - switch.turn_on: relay_2_on
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x02, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            then:
+              - switch.turn_on: relay_2_on
+            else:
+              - switch.turn_on: relay_12_on
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x03, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            condition:
+              switch.is_off: relay_1
+            then:
+              - switch.turn_on: relay_12_off
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x00, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+            then:
+              - switch.turn_on: relay_12_off
+            else:
+              - switch.turn_on: relay_1_on
```
```suggestion
              - uart.write: [0xA0, 0x04, 0x01, 0xA1]
```

OttoWinter commented on cookbook/sonoff-dual-light-switch.rst:
```diff
@@ -0,0 +1,270 @@
+    wifi:
+      ssid: !secret wifi_ssid
+      password: !secret wifi_password
+      fast_connect: True
```
Is there a reason for these fast connects? I would prefer not to have them here, as fast_connect disables some functionality like round-robin wifi network selection that makes wifi more stable.

bensuffolk replied:
Its because my HASS wifi is hidden, so its required, and I just copied my files. Clearly not needed for an example, and I shall remove them

OttoWinter reviewed (commented):
Hi, thanks for updating the cookbook entry.

The text looks great now! There are just some notes about the YAML - some things can be simplified.

bensuffolk committed:
Update cookbook/sonoff-basic-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
Update cookbook/sonoff-dual-light-switch.rst

Co-Authored-By: Otto Winter <otto@otto-winter.com>

bensuffolk committed:
YAML updates

Removed fast_connect, UART swotches from Dual R1, and reference to an issue that no longer exists

bensuffolk commented:
I have removed the fast_connect lines, and the now not required UART switches from the R2, so hopefully we are good to go.
